2023.2 ProcedureCompilation issues

### Issue description

I've imported my working mod from the 2023.1 version to 2023.2 snapshot and get compilation errors with 1.19.2 and 1.19.4 builds.
C:\MCreator-Releases\MCreator20232\workspaces\gravels\src\main\java\net\mcreator\tmtmcgravels\procedures\CompletedAllGravelsProProcedure.java:51: error: illegal attempt to redefine an existing match binding
&& _plr.getAdvancements().getOrStartProgress(_plr.server.getAdvancements().getAdvancement(new ResourceLocation("tmtmcgravels:gravel_light_gray_ach"))).isDone() && entity instanceof ServerPlayer _plr

C:\MCreator-Releases\MCreator20232\workspaces\gravels\src\main\java\net\mcreator\tmtmcgravels\procedures\CompletedAllGravelsProProcedure.java:53: error: variable _plr is already defined in method execute(Event,Entity)
if (entity instanceof ServerPlayer _plr && _plr.level instanceof ServerLevel

### How to reproduce this issue?

Imported the workspace ZIP file from 2023.1 to 2023.2 and after compilation was done, it though errors on the 6 Completion Procedures that check for existing achievements and then grants XP or XP Levels if all are done.
![image](https://user-images.githubusercontent.com/10309346/236527802-d4420b85-816e-437c-a812-2820b10a64d6.png)
![image](https://user-images.githubusercontent.com/10309346/236527952-051e2f38-a26b-48e9-a028-ad7b27545ee6.png)
I've tried moving each group of two to a separate If/do block and that had no change in behavior.  I've even tried double clicking on each of the advancement checks and re-selected the same Advancement but no difference in behavior.
Here is example of one of the references:
![image](https://user-images.githubusercontent.com/10309346/236528757-7bd17806-2612-46fa-b26a-dabcd66ea496.png)


### Operating system

Windows

### Details

MCreator 2023.2 Snapshot
Forge 1.19.2 and 1.194

### Example workspace

_No response_

### Logs

https://pastebin.com/PVWGcFzf

### Issue tracker rule checks (please read carefully)

- [X] I have checked that my problem <a href='https://github.com/MCreator/MCreator/issues?q=is%3Aissue' target='_blank'>is not already reported</a>
- [X] I have checked that my problem is not covered on <a href='https://mcreator.net/support/knowledgebase' target='_blank'>Knowledge Base</a> or on <a href='https://mcreator.net/wiki' target='_blank'>MCreator's Wiki</a>
- [X] I have checked that my written content does not violate the <a href='https://mcreator.net/wiki/general-publishing-guidelines' target='_blank'>publishing guidelines</a>

--- a/plugins/generator-1.19.2/forge-1.19.2/procedures/entity_has_achievement.java.ftl
+++ b/plugins/generator-1.19.2/forge-1.19.2/procedures/entity_has_achievement.java.ftl
@@ -1,2 +1,3 @@
-(${input$entity} instanceof ServerPlayer _plr && _plr.level instanceof ServerLevel && _plr.getAdvancements()
-        .getOrStartProgress(_plr.server.getAdvancements().getAdvancement(new ResourceLocation("${generator.map(field$achievement, "achievements")}"))).isDone())
+<#assign plr = "_plr" + customBlockIndex>
+(${input$entity} instanceof ServerPlayer ${plr} && ${plr}.level instanceof ServerLevel && ${plr}.getAdvancements()
+        .getOrStartProgress(${plr}.server.getAdvancements().getAdvancement(new ResourceLocation("${generator.map(field$achievement, "achievements")}"))).isDone())

--- a/plugins/generator-1.19.2/forge-1.19.2/procedures/world_data_isday.java.ftl
+++ b/plugins/generator-1.19.2/forge-1.19.2/procedures/world_data_isday.java.ftl
@@ -1,1 +1,1 @@
-(world instanceof Level _lvl && _lvl.isDay())
+(world instanceof Level _lvl${customBlockIndex} && _lvl${customBlockIndex}.isDay())

--- a/plugins/generator-1.19.2/forge-1.19.2/procedures/world_data_israided.java.ftl
+++ b/plugins/generator-1.19.2/forge-1.19.2/procedures/world_data_israided.java.ftl
@@ -1,2 +1,2 @@
 <#include "mcelements.ftl">
-(world instanceof ServerLevel _level && _level.isRaided(${toBlockPos(input$x,input$y,input$z)}))
+(world instanceof ServerLevel _level${customBlockIndex} && _level${customBlockIndex}.isRaided(${toBlockPos(input$x,input$y,input$z)}))

--- a/plugins/generator-1.19.2/forge-1.19.2/procedures/world_data_isvillage.java.ftl
+++ b/plugins/generator-1.19.2/forge-1.19.2/procedures/world_data_isvillage.java.ftl
@@ -1,2 +1,2 @@
 <#include "mcelements.ftl">
-(world instanceof ServerLevel _level && _level.isVillage(${toBlockPos(input$x,input$y,input$z)}))
+(world instanceof ServerLevel _level${customBlockIndex} && _level${customBlockIndex}.isVillage(${toBlockPos(input$x,input$y,input$z)}))

--- a/plugins/generator-1.19.4/forge-1.19.4/procedures/advancement_isequalto.java.ftl
+++ b/plugins/generator-1.19.4/forge-1.19.4/procedures/advancement_isequalto.java.ftl
@@ -1,2 +1,2 @@
-(world instanceof Level _lvl && _lvl.getServer() != null && _lvl.getServer().getAdvancements()
+(world instanceof Level _lvl${customBlockIndex} && _lvl${customBlockIndex}.getServer() != null && _lvl${customBlockIndex}.getServer().getAdvancements()
     .getAdvancement(new ResourceLocation("${generator.map(field$achievement, "achievements")}" )).equals(advancement))

--- a/plugins/generator-1.19.4/forge-1.19.4/procedures/blockat_plant_type.java.ftl
+++ b/plugins/generator-1.19.4/forge-1.19.4/procedures/blockat_plant_type.java.ftl
@@ -1,3 +1,3 @@
 <#include "mcelements.ftl">
-(world.getBlockState(${toBlockPos(input$x,input$y,input$z)}).getBlock() instanceof IPlantable _plant &&
-	_plant.getPlantType(world, ${toBlockPos(input$x,input$y,input$z)}) == PlantType.${generator.map(field$planttype, "planttypes")})
+(world.getBlockState(${toBlockPos(input$x,input$y,input$z)}).getBlock() instanceof IPlantable _plant${customBlockIndex} &&
+	_plant${customBlockIndex}.getPlantType(world, ${toBlockPos(input$x,input$y,input$z)}) == PlantType.${generator.map(field$planttype, "planttypes")})

--- a/plugins/generator-1.19.4/forge-1.19.4/procedures/entity_check_creature_type.java.ftl
+++ b/plugins/generator-1.19.4/forge-1.19.4/procedures/entity_check_creature_type.java.ftl
@@ -1,1 +1,1 @@
-(${input$entity} instanceof LivingEntity _livEnt && _livEnt.getMobType() == MobType.${field$type})
+(${input$entity} instanceof LivingEntity _livEnt${customBlockIndex} && _livEnt${customBlockIndex}.getMobType() == MobType.${field$type})

--- a/plugins/generator-1.19.4/forge-1.19.4/procedures/entity_has_achievement.java.ftl
+++ b/plugins/generator-1.19.4/forge-1.19.4/procedures/entity_has_achievement.java.ftl
@@ -1,2 +1,3 @@
-(${input$entity} instanceof ServerPlayer _plr && _plr.level instanceof ServerLevel && _plr.getAdvancements()
-        .getOrStartProgress(_plr.server.getAdvancements().getAdvancement(new ResourceLocation("${generator.map(field$achievement, "achievements")}"))).isDone())
+<#assign plr = "_plr" + customBlockIndex>
+(${input$entity} instanceof ServerPlayer ${plr} && ${plr}.level instanceof ServerLevel && ${plr}.getAdvancements()
+        .getOrStartProgress(${plr}.server.getAdvancements().getAdvancement(new ResourceLocation("${generator.map(field$achievement, "achievements")}"))).isDone())

--- a/plugins/generator-1.19.4/forge-1.19.4/procedures/entity_has_screen_open.java.ftl
+++ b/plugins/generator-1.19.4/forge-1.19.4/procedures/entity_has_screen_open.java.ftl
@@ -1,1 +1,1 @@
-(${input$entity} instanceof Player _plr && _plr.containerMenu instanceof ${field$guiname}Menu)
+(${input$entity} instanceof Player _plr${customBlockIndex} && _plr${customBlockIndex}.containerMenu instanceof ${field$guiname}Menu)

--- a/plugins/generator-1.19.4/forge-1.19.4/procedures/entity_haspotioneffect.java.ftl
+++ b/plugins/generator-1.19.4/forge-1.19.4/procedures/entity_haspotioneffect.java.ftl
@@ -1,1 +1,1 @@
-(${input$entity} instanceof LivingEntity _livEnt && _livEnt.hasEffect(${generator.map(field$potion, "effects")}))
+(${input$entity} instanceof LivingEntity _livEnt${customBlockIndex} && _livEnt${customBlockIndex}.hasEffect(${generator.map(field$potion, "effects")}))

--- a/plugins/generator-1.19.4/forge-1.19.4/procedures/item_can_smelt.java.ftl
+++ b/plugins/generator-1.19.4/forge-1.19.4/procedures/item_can_smelt.java.ftl
@@ -1,2 +1,2 @@
 <#include "mcitems.ftl">
-(world instanceof Level _lvlCanSmelt && _lvlCanSmelt.getRecipeManager().getRecipeFor(RecipeType.SMELTING, new SimpleContainer(${mappedMCItemToItemStackCode(input$item, 1)}), _lvlCanSmelt).isPresent())
+(world instanceof Level _level${customBlockIndex} && _level${customBlockIndex}.getRecipeManager().getRecipeFor(RecipeType.SMELTING, new SimpleContainer(${mappedMCItemToItemStackCode(input$item, 1)}), _level${customBlockIndex}).isPresent())

--- a/plugins/generator-1.19.4/forge-1.19.4/procedures/world_data_is_block_powered.java.ftl
+++ b/plugins/generator-1.19.4/forge-1.19.4/procedures/world_data_is_block_powered.java.ftl
@@ -1,2 +1,2 @@
 <#include "mcelements.ftl">
-(world instanceof Level _lvl_isPow && _lvl_isPow.hasNeighborSignal(${toBlockPos(input$x,input$y,input$z)}))
+(world instanceof Level _level${customBlockIndex} && _level${customBlockIndex}.hasNeighborSignal(${toBlockPos(input$x,input$y,input$z)}))

--- a/plugins/generator-1.19.4/forge-1.19.4/procedures/world_data_isday.java.ftl
+++ b/plugins/generator-1.19.4/forge-1.19.4/procedures/world_data_isday.java.ftl
@@ -1,1 +1,1 @@
-(world instanceof Level _lvl && _lvl.isDay())
+(world instanceof Level _lvl${customBlockIndex} && _lvl${customBlockIndex}.isDay())

--- a/plugins/generator-1.19.4/forge-1.19.4/procedures/world_data_israided.java.ftl
+++ b/plugins/generator-1.19.4/forge-1.19.4/procedures/world_data_israided.java.ftl
@@ -1,2 +1,2 @@
 <#include "mcelements.ftl">
-(world instanceof ServerLevel _level && _level.isRaided(${toBlockPos(input$x,input$y,input$z)}))
+(world instanceof ServerLevel _level${customBlockIndex} && _level${customBlockIndex}.isRaided(${toBlockPos(input$x,input$y,input$z)}))

--- a/plugins/generator-1.19.4/forge-1.19.4/procedures/world_data_isvillage.java.ftl
+++ b/plugins/generator-1.19.4/forge-1.19.4/procedures/world_data_isvillage.java.ftl
@@ -1,2 +1,2 @@
 <#include "mcelements.ftl">
-(world instanceof ServerLevel _level && _level.isVillage(${toBlockPos(input$x,input$y,input$z)}))
+(world instanceof ServerLevel _level${customBlockIndex} && _level${customBlockIndex}.isVillage(${toBlockPos(input$x,input$y,input$z)}))

